main.go

```diff
@@ -49,7 +32,6 @@ var upgrader = websocket.Upgrader{
 func socketHandler(w http.ResponseWriter, r *http.Request) {
 	conn, err := upgrader.Upgrade(w, r, nil)
 	defer conn.Close()
```

Comment:
This looks like a bug. `conn.Close()` should come after the error check as I expect `conn` will be `nil` in this case. This becomes noticeable when you remove that blank line to cuddle the error check with the call that returns the error, but there is this defer in the middle.

As an aside, argghhhhhh! https://pkg.go.dev/github.com/gorilla/websocket#Upgrader.Upgrade is the docs for that call. I wanted it to tell me what it returns. Notice how the text says nothing at all about return values - you just have to assume. This is the style of bad docs I talked about with microsoft docs.

Reply:
yeah, right you are. Updated.
I wonder also if we should make and httpe.HandlerFuncE out of it. Would make it quite a bit shoreter.

Reply:
I wouldn't bother yet. it's not that long as it is, and will just remove a few `return`s. Wait until it is a real thing,